pkg/reconciler/route/resources/certificate.go

```diff
@@ -52,7 +52,7 @@ func MakeCertificate(owner kmeta.OwnerRefableAccessor, ownerLabelKey string, dns
 		},
 		Spec: networkingv1alpha1.CertificateSpec{
 			DNSNames:   []string{dnsName},
-			SecretName: certName,
+			SecretName: secretName,
```

Comment:
The godoc for this says
> SecretName is the name of the secret resource to *_store_* the SSL certificate in.

It's not obvious whether different controllers will check this secret and not overwrite it. Even if it does not what happens when the certificate in the secret expires?

I wonder if when we bring your own certificate we don't use the default certificate class.

Reply:
> I wonder if when we bring your own certificate we don't use the default certificate class.

I guess we still need something to set the certificate as 'ready`

Reply:
At least in net-http01 I see it checks for the secret before trying to create it [here](https://github.com/knative-sandbox/net-http01/blob/4a6ac48a968cfc7510b96dd592a91453551a6593/pkg/reconciler/certificate/certificate.go#L64) which is consistent with what I've tested, this also happens with cert-manager but I haven't found the specific code that does this in cert-manager. I'm not sure if this answers your question

Reply:
`net-http01` may be work well but we can't assume other's won't

ie. cert manager

The secret name is [delegated to a cert manager CRD here](https://github.com/knative-sandbox/net-certmanager/blob/8ef52bb055cabd2896e6b4103a1d31246b4c4cd2/pkg/reconciler/certificate/resources/cert_manager_certificate.go#L41-L46). A rough scan looks like there are edge cases where it might fail

Specifically the secret is evaluated against the[ following policy chain](https://github.com/jetstack/cert-manager/blob/c7514d92624613395eb00da8594ffa5391a0b169/pkg/controller/certificates/trigger/policies/policies.go#L75-L81) to determine whether to re-issue the certificate 


Two concerning assertions are:

#### SecretPrivateKeyMatchesSpec
When creating the cert manager crd we don't indicate the private key type. When it's not set cert manager assumes we're bringing a RSA private key. Thus if we brought a `ECDSA` cert then I think it trigger a reissue that'll fail/overwrite our secret

#### SecretIssuerAnnotationsNotUpToDate

it seems like the secret needs the 'correct' annotations or it'll be re-issued - these annotations are cert manager specific.